src/super_gradients/convert_recipe_to_code.py

```diff
@@ -57,45 +57,125 @@ def try_import_black():
             object_name = f"{prefix}".replace(".", "_").lower()
             objects[object_name] = (target_class, target_params)
-            cfg = object_name
+            import_statement = extract_import_statement_from_hydra_target(target_class)
+            cfg = _wrap_variable_reference(object_name)
+            additional_imports.append(import_statement)
 
     elif isinstance(cfg, ListConfig):
         for index, item in enumerate(cfg):
-            item, objects = recursively_walk_and_extract_hydra_targets(item, objects, prefix=f"{prefix}_{index}")
+            item, objects, imports = recursively_walk_and_extract_hydra_targets(item, objects, prefix=f"{prefix}_{index}")
+            additional_imports.extend(imports)
             cfg[index] = item
     else:
         pass
-    return cfg, objects
 
+    return _unwrap_variable_references(cfg), objects, list(set(additional_imports))
+
+
+def _wrap_variable_reference(variable_name: str) -> Tuple[str, str]:
+    """
+    Wrap a variable name with a tuple to indicate that it is a reference to a variable.
+    It is used to discriminate between "normal" strings and references to instantiated objects.
+    We need to be able to distinguish between them because for strings we need to wrap them in quotes in the generated script.
+    :param variable_name: Name of the variable
+    :return:              A tuple that indicates that this is a reference to a variable
+    """
+    return "__reference__", variable_name
+
+
+def _unwrap_variable_references(cfg: DictConfig) -> DictConfig:
+    """
+    Unwrap variable references from the config. This function modify the configuration in-place.
+    This method replace variable reference entries: ("__reference__", object_name) -> object_name
+    """
+    if isinstance(cfg, DictConfig):
+        for key, value in cfg.items():
+            cfg[key] = _unwrap_variable_references(value)
+    elif isinstance(cfg, (ListConfig, tuple)) and len(cfg) == 2 and cfg[0] == "__reference__":
+        return cfg[1]
+    elif isinstance(cfg, ListConfig):
+        for index, item in enumerate(cfg):
+            cfg[index] = _unwrap_variable_references(item)
+
+    return cfg
+
+
+def extract_import_statement_from_hydra_target(path: str) -> str:
+    """
+    Locate an object by name or dotted path and return the import statement for it.
+    If a hydra target is "numpy.arange" then in order to instantiate it via code we should
+    import "numpy" (top-level path) and this method extracts root path and returns it.
+
+    Most of the logic borrowed from hydra._internal.utils._locate
+    """
+    if path == "":
+        raise ImportError("Empty path")
+    from importlib import import_module
+
+    parts = path.split(".")
+    for part in parts:
+        if not len(part):
+            raise ValueError(f"Error loading '{path}': invalid dotstring. Relative imports are not supported.")
+    assert len(parts) > 0
```

Comment:
Minor, but why do we need this ? If it fails, the error would be super unintuitive for the user. So I assume it should never be true and is here more as self-documenting ?

Reply:
Basically it was like that in hydra. 
It is to prevent user attempt to use `_target_: .something.foo` or import from parent (`..something.else`)

Reply:
Mmmhh but if it happens the user would get a cryptic error, no ?
Wouldnt it be better to at least write this in the assert error message ?